db/builtin_routine.hh

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include "builtin_routine_registry.hh"
+#include <seastar/core/shared_ptr.hh>
+namespace db {
+// An execution context of a builtin stored procedure, e.g.
+// system.repair().
+// Such procedures are statically defined in the code and invoked
+// by CALL statement if the name matches.
+struct builtin_routine {
+    using parameter_map = std::unordered_map<sstring, sstring>;
+    using output = std::vector<std::vector<data_value>>;
+    using output_metadata = std::vector<std::pair<sstring, data_type>>;
+
+    std::function<future<output>(parameter_map&&)> routine;
```

Comment:
It's safer for coroutine routines to pass by value, since pass-by-reference will not copy the parameter. It's true that a function that accepts by value can be assigned to function<... (parameter_map&&)>, but people may copy the signature and fall into a trap.